### PR TITLE
upgraded to dp-kafka v3 and healthcheck subscription

### PR DESCRIPTION
**Pull request by DavidSubiros**
### What

Stop consuming Kafka messages on unhealthy state, and start again on healthy state. Trello card: https://trello.com/c/TrKac6Lm/5351-start-stop-consuming-on-healthy-unhealthy-for-dp-cantabular-csv-exporter

- dp-healthcheck has been upgraded to version 1.2.0, which implements a subscirbe/notify mechanism for health states (code level subscirbes)
- dp-kafka has been upgraded to version 3, which implements the health callback - starting consuming messages on healthy state, and stop consuming messages on unhealhty state.
- kafka consumer is subscribed to the necessary Checks on healthcheck
- Added feature flag to enable/disable this behaviour - if the behaviour is disabled, start the kafka consumer straight away instead of subscribing it to health updates.
- removed consumer loop and related code/structs, and use the new handler register functionality implemented in dp-kafka (simplifying the code substantially in the service)
- renamed `common-output` var names have been replaced for `csv-created`, to match the topic name being produced
- reintroduced config logging in main.go

### How to review

- Make sure unit/component tests pass
- (optional - I did test it locally) - You can see the new functionality by removing the `minIO` docker container and see that it stops consuming, and it starts again when you add it back. In dp-compose/cantabular-imports you can do:
  - (1) start the cantabular docker-compose (or at least the back end)
  - (2) track dp-cantabular-csv-expoter logs: `./logs dp-cantabular-csv-exporter`
  - (3) wait until you see that the sarama session has been established (expected log message: `kafka consumer group is starting` followed by `kafka consumer group is consuming: sarama consumer group session setup ok: a new go-routine will be created for each partition assigned to this consumer`)
  - (4) kill the minIO container(s) - I had to kill 2 containers (`docker kill <id1> <id2>`)
  - (5) in dp-cantabular-csv-exporter logs, see that in a few seconds the sarama session is terminated (expected message `kafka consumer group has finished consuming: sarama consumer group session cleanup finished: all go-routines have completed` followed by `kafka consumer group is stopped` - so that we know the reason is that the consumer has been stopped)
  - (6) run an import (you may use the helper):
    - `make init-db` 
    - `cd helpers`
    - `./start-import.sh`
  - (7) check that the import is `completed` and the instance is created, but no download is available
  - (8) start the minio container(s) with the same IDs that were used in step 4 (`docker start <id1> <id2>`)
  - check that after some time `cantabular-csv-exporter` logs the start event and sarama session established (same messages as in step 3), and then it consumes the `instance-complete` events straight away (that had been generated while the service was unhealthy)

### Who can review

Anyone

### CODE REVIEW

File: event/event.go [hunk 1]
```diff
@@ -6,9 +6,8 @@ type InstanceComplete struct {
 	CantabularBlob string `avro:"cantabular_blob"`
 }
 
-// CommonOutputCreated provides an avro structure for an Output Created event
-type CommonOutputCreated struct {
-	FilterID   string `avro:"filter_output_id"`
+// CsvCreated provides an avro structure for a CSV Created event
+type CsvCreated struct {
```
Comment (ChrysmOre) [inline event/event.go:10]: This name should probably be capitalised as in CSVCreated

File: main.go [hunk 2]
```diff
@@ -34,32 +35,36 @@ func main() {
...
 	svcErrors := make(chan error, 1)
 
 	// Read config
 	cfg, err := config.Get()
 	if err != nil {
 		return fmt.Errorf("unable to retrieve service configuration: %w", err)
 	}
+	log.Info(ctx, "config on startup", log.Data{"config": cfg, "build_time": BuildTime, "git-commit": GitCommit})
+
+	serviceCtx, cancel := context.WithCancel(ctx)
```
Comment (ChrysmOre) [inline main.go:48]: I wonder if it's possible to reuse the ctx variable like `ctx, cancel = context.Cancel(ctx)` here, the way you might do `i = i +1`?

File: event/event.go (hunk 1 above)
Comment (DavidSubiros) [reply event/event.go:10]: Also renaming the corresponding schema.

File: main.go (hunk 2 above)
Comment (DavidSubiros) [reply main.go:48]: The intention was to only use the `serviceCtx` for the service. But `ctx` is only used for logging, so we could as well rename it as you mention, and use it for everything

**Review by ChrysmOre — APPROVED** [review]

**Pull request merged**